### ISSUE TITLE
"dead" Backticks and Tilde on Scandinavian keyboards

### ISSUE THREAD

**Issue by haakonstorm** [pqrs-org/Karabiner-Elements#2023]
# Continuation of #1205 

- Stackoverflow link from that thread was nixed and can't get this to work properly. I've been working on this for many months.

First, thank you so much for Karabiner. It is an awesome tool, (albeit a little difficult to configure). 

I'm on a scandiavian keyboard (Norwegian - macOS 10.15 - Apple Magic Keyboard (non-keypad, but affects all macos versions and all Mac keyboards. 

## Backtick
K-Eventviewer outputs `code 46 equal_sign` when hitting they backtick key alone.
When pressed alone, backtick is a dead key.  To get backtick, you must use `option` + `backtick` (and in these days with so much Markdown around, it's getting increasingly painful!)

Eventviewer outputs `key_down` `46` `equal_sign` `flags: left_alt` on hitting `option` + `backtick` simultaneously, in addition to the modifiers up/down: (And why do everything appear twice? I have a Logitech G Pro USB-Mouse (with Logitech G Hub software):

<img width="512" alt="image" src="https://user-images.githubusercontent.com/193781/64421674-6a682400-d0a2-11e9-94b1-842ffd89296f.png">

I can't find any *complex modifiers* to *exchange* these two 'functions' on the same physical key. Is it possible?

## Tilde

It is even worse getting to type `tilde`. The `tilde` character is on a key the eventviewer reports as `48 close bracket`, also a dead key. Hitting `option` + `tillde` is a third dead key. To get tilde, you must use `option` + `tilde` and then hit space. *carpal tunnel song playing in the far distance*

# Questions
1. Did anyone successfully create  some sort of modifier or other solution to remap these dead keys somehow, to make them dead only with modifier or otherwise if the dead keys must continue to be ... dead?

2. Is it possible to create more dead keys somehow, like some sort of vim-mode, on Scandinavian keyboards?

## Extra: config
```
  "input_source": {
        "input_source_id": "com.apple.keylayout.Norwegian",
        "language": "nb"
    },
    "system_preferences_properties": {
        "keyboard_types": [
            [
                {
                    "country_code": 0,
                    "product_id": 49285,
                    "vendor_id": 1133
                },
                40
            ],
            [
                {
                    "country_code": 0,
                    "product_id": 10203,
                    "vendor_id": 5824
                },
                40
            ]
        ],
        "scroll_direction_is_natural": true,
        "use_fkeys_as_standard_function_keys": true
    },
    "variables": {},
    "virtual_hid_keyboard_country_code": 0,
    "virtual_hid_keyboard_keyboard_type": "ansi"
}
```

**Comment by haakonstorm** [comment]
...Anyone?

**Comment by jonasson2** [comment]
This works on Icelandic keyboard (where the backtick is dead and on a key reported as "backslash" by the event viewer and obtained with the option key):

`                      "description": "Backtick add space",
                        "manipulators": [
                            {
                                "from": {
                                    "key_code": "backslash",
                                    "modifiers": {
                                        "mandatory": [
                                            "right_option"
                                        ]
                                    }
                                },
                                "to": [
                                    {
                                      "key_code": "backslash",
                                      "modifiers": ["right_option"]
                                    },
                                  {"key_code": "spacebar"}
                                ],
                                "type": "basic"
                            }
                        ]
`

Basically the key is mapped to itself, but with a subsequent space (sorry, no line breaks appear, I guess because this is a comment)

**Comment by busla** [comment]
> This works on Icelandic keyboard (where the backtick is dead and on a key reported as "backslash" by the event viewer and obtained with the option key):
> 
> ` "description": "Backtick add space", "manipulators": [ { "from": { "key_code": "backslash", "modifiers": { "mandatory": [ "right_option" ] } }, "to": [ { "key_code": "backslash", "modifiers": ["right_option"] }, {"key_code": "spacebar"} ], "type": "basic" } ]`
> 
> Basically the key is mapped to itself, but with a subsequent space (sorry, no line breaks appear, I guess because this is a comment)

brilliant!

**Comment by busla** [comment]
Extended the `spacebar` hack from @jonasson2 to get consistent `backtick` and `tilde` characters regardless of keyboard input (I write code with US input and use Icelandic input when communicating in Icelandic).

```
{
  "title": "Fixed Backtick and Tilde Mapping for Icelandic Layout",
  "rules": [
    {
      "description": "Map non_us_backslash to backtick and tilde (Icelandic layout only)",
      "manipulators": [
        {
          "type": "basic",
          "from": {
            "key_code": "non_us_backslash",
            "modifiers": {
              "mandatory": ["right_shift"]
            }
          },
          "to": [
            {
              "key_code": "close_bracket",
              "modifiers": ["left_option"]
            }
          ],
          "conditions": [
            {
              "type": "input_source_if",
              "input_sources": [
                {
                  "input_source_id": "com.apple.keylayout.Icelandic"
                }
              ]
            }
          ]
        },
        {
          "type": "basic",
          "from": {
            "key_code": "non_us_backslash",
            "modifiers": {
              "optional": ["any"]
            }
          },
          "to": [
            {
              "key_code": "backslash",
              "modifiers": ["left_option"]
            },
            {
              "key_code": "spacebar"
            }
          ],
          "conditions": [
            {
              "type": "input_source_if",
              "input_sources": [
                {
                  "input_source_id": "com.apple.keylayout.Icelandic"
                }
              ]
            }
          ]
        }
      ]
    }
  ]
}
```
